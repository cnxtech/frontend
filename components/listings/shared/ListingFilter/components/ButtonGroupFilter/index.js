import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import Button from '@emcasa/ui-dom/components/Button'
import FilterButton from '../FilterButton'

class ButtonGroupFilter extends PureComponent {
  render() {
    const { initialValue, userValue, onChange, values } = this.props
    return (
      <Button.Group
        flexWrap="wrap"
        initialValue={initialValue}
        strategy={{
          isSelected: (selectedValue, value) => selectedValue === value,
          update: (selectedValue, value) => (selectedValue === value ? null : value)
        }}
        onChange={(value) => {
          if (value === null) {
            return
          }
          let values = {minValue: value}
          if (value === 0) {
            values.maxValue = 0
          }
          onChange(values)
      }}>
        {values.map((item, index) => {
          return (
            <FilterButton
              key={index}
              active={userValue === item.value}
              value={item.value}
            >
              {item.label}
            </FilterButton>
          )
        })}
      </Button.Group>
    )
  }
}

ButtonGroupFilter.propTypes = {
  initialValue: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
  values: PropTypes.array.isRequired,
  userValue: PropTypes.number
}

/**
 * values: [
 *  {
 *    value: 0,
 *    label: 'Sem vagas'
 *   },
 *   {
 *     value: 1,
 *     label: '1'
 *   }
 *   ...
 * ]
 */

export default ButtonGroupFilter
