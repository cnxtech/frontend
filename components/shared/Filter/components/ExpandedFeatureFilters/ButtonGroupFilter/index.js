import React from 'react'
import styled from 'styled-components'
import {themeGet} from 'styled-system'
import Button from '@emcasa/ui-dom/components/Button'

const ButtonGroup = styled(function ButtonGroup({
  children,
  style,
  className,
  strategy,
  currentValue,
  onChange
}) {
  return (
    <Button.Group
      style={style}
      className={className}
      strategy={strategy}
      selectedValue={currentValue}
      onChange={onChange}
    >
      {React.Children.map(children, (element) =>
        React.cloneElement(element, {
          className: 'filterButton',
          type: 'button'
        })
      )}
    </Button.Group>
  )
})`
  margin-left: -${themeGet('space.2')}px;
  margin-bottom: -${themeGet('space.2')}px;
  flex-wrap: wrap;
  .filterButton {
    margin-left: ${themeGet('space.2')}px;
    margin-bottom: ${themeGet('space.2')}px;
  }
`

export default ButtonGroup
