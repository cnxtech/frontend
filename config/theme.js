import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
