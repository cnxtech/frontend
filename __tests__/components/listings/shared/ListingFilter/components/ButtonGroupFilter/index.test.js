import React from 'react'
import ButtonGroupFilter from 'components/listings/shared/ListingFilter/components/ButtonGroupFilter'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import theme from 'config/theme'
import { ThemeProvider } from 'styled-components'

describe('<ButtonGroupFilter/>', () => {
  it('should render component', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <ButtonGroupFilter
          initialValue={0}
          userValue={1}
          onChange={(value) => {}}
          values={[
            {value: 0, label: 'Sem vagas'},
            {value: 1, label: '1'},
            {value: 2, label: '2'},
            {value: 3, label: '3'},
            {value: 4, label: '4'},
            {value: 5, label: '+'},
          ]}
        />
      </ThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
