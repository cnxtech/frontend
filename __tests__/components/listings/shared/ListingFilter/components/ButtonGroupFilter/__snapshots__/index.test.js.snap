// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroupFilter/> should render component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  outline: none;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 4px;
  border-width: 1px;
  width: auto;
  height: 40px;
  margin-bottom: 10px;
  margin-right: 10px;
  padding-left: 10px;
  padding-right: 10px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c2 :hover {
  background-color: #ffffff;
}

.c6 {
  outline: none;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #f50057;
  background-color: #f50057;
  border-radius: 4px;
  border-width: 1px;
  width: auto;
  height: 40px;
  margin-bottom: 10px;
  margin-right: 10px;
  padding-left: 10px;
  padding-right: 10px;
  color: #ffffff;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c6 :hover {
  background-color: #f50057;
}

.c1 {
  border: 1px solid #f50057;
  color: #f50057;
  font-size: 14px;
  margin: 0 10px 4px 0;
  height: 32px;
}

.c1:last-child {
  margin-right: 0;
}

.c5 {
  border: 1px solid #f50057;
  color: #ffffff;
  font-size: 14px;
  margin: 0 10px 4px 0;
  height: 32px;
}

.c5:last-child {
  margin-right: 0;
}

.c3 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

@media screen and (min-width:780px) {
  .c2 {
    padding-left: 14px;
    padding-right: 14px;
  }
}

@media screen and (min-width:780px) {
  .c6 {
    padding-left: 14px;
    padding-right: 14px;
  }
}

<div
  className="c0"
  onChange={[Function]}
>
  <button
    className="c1 c2"
    fontSize="default"
    height="medium"
    onClick={[Function]}
  >
    <div
      className="c3 c4"
    >
      Sem vagas
    </div>
  </button>
  <button
    className="c5 c6"
    fontSize="default"
    height="medium"
    onClick={[Function]}
  >
    <div
      className="c3 c4"
    >
      1
    </div>
  </button>
  <button
    className="c1 c2"
    fontSize="default"
    height="medium"
    onClick={[Function]}
  >
    <div
      className="c3 c4"
    >
      2
    </div>
  </button>
  <button
    className="c1 c2"
    fontSize="default"
    height="medium"
    onClick={[Function]}
  >
    <div
      className="c3 c4"
    >
      3
    </div>
  </button>
  <button
    className="c1 c2"
    fontSize="default"
    height="medium"
    onClick={[Function]}
  >
    <div
      className="c3 c4"
    >
      4
    </div>
  </button>
  <button
    className="c1 c2"
    fontSize="default"
    height="medium"
    onClick={[Function]}
  >
    <div
      className="c3 c4"
    >
      +
    </div>
  </button>
</div>
`;
